README.md

```diff
@@ -66,6 +66,23 @@ See [flake.parts](https://flake.parts/options.html)
 
  -  `getSystem`: function from system string to the `config` of the appropriate `perSystem`.
 
+ -  `moduleWithSystem`: function that brings the `perSystem` module arguments.
+    This allows a module to reference the defining flake without introducing
+    global variables that need to be set by the user of the module.
+
+    ```nix
+    { moduleWithSystem, ... }:
+    {
+      nixosModules = moduleWithSystem (
```

Comment:
Oops :)

Just noting: It's acceptable to define host configurations as modules, but typically you'd only define `nixosConfigurations` for that use case.

I'm going with `nixosModules.default`, the idea being that this is the "`foo`" flake defining its only service.

Reply:
`nixosConfigurations` presumably don't suffer from global variables as much, because they're not intended to be extended or reused, so `specialArgs` and `_module.args` don't _technically_ need to be avoided in modules that are only used locally. However, we _should_ avoid them, to shorten the learning curve and save some frustration.